sources/packages/backend/apps/api/src/route-controllers/restriction/_tests_/e2e/restriction.institution.controller.getStudentRestrictionDetail.e2e-spec.ts

```diff
@@ -0,0 +1,154 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import {
+  createFakeInstitutionLocation,
+  saveFakeStudent,
+  saveFakeStudentRestriction,
+  createFakeApplication,
+} from "@sims/test-utils";
+import { DataSource, Repository } from "typeorm";
+import {
+  Institution,
+  InstitutionLocation,
+  Application,
+  RestrictionNotificationType,
+} from "@sims/sims-db";
+
+import {
+  authorizeUserTokenForLocation,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+  InstitutionTokenTypes,
+} from "../../../../testHelpers";
+import * as request from "supertest";
+import { getUserFullName } from "../../../../utilities";
+
+describe("RestrictionInstitutionController(e2e)-getStudentRestrictions", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let collegeF: Institution;
+  let collegeC: Institution;
+  let collegeFLocation: InstitutionLocation;
+  let collegeCLocation: InstitutionLocation;
+  let applicationRepo: Repository<Application>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+
+    const { institution } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeF = institution;
+    collegeFLocation = createFakeInstitutionLocation(collegeF);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+    applicationRepo = appDataSource.getRepository(Application);
+  });
+
+  it("Should return forbidden 403 error code when a non BC Public Institution User tries to get student restriction by student restriction id.", async () => {
+    // Arrange
+    const { institution } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeC = institution;
+    collegeCLocation = createFakeInstitutionLocation(collegeC);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+    const student = await saveFakeStudent(appDataSource);
+    const application = createFakeApplication({
+      location: collegeCLocation,
+      student,
+    });
+    await applicationRepo.save(application);
+    const studentRestriction = await saveFakeStudentRestriction(
+      appDataSource,
+      { student, application, creator: student.user },
```

Comment:
I would say that in saveFakeStudentRestriction, the creator can be set by default and not required to be passed from here.

```ts
studentRestriction.creator = relations?.creator?? createFakeUser();
```